### ISSUE TITLE
Docker container crashed unexpectedly

### ISSUE THREAD

**Issue by vdv73rus** [docker/for-mac#676]
### Expected behavior

Docker container works fine
### Actual behavior

Docker contatiner crashes after several minutes
### Information

Docker for Mac: version: 1.12.1 (2d5b4d9)
OS X: version 10.11.6 (build: 15G1004)
logs: /tmp/31AFFFAA-CBCC-40E4-8F3D-707D54630FD6/20160920-153729.tar.gz
[OK]     docker-cli
[OK]     virtualization kern.hv_support
[OK]     menubar
[OK]     moby-syslog
[OK]     dns
[OK]     disk
[OK]     system
[OK]     app
[OK]     osxfs
[OK]     virtualization VT-X
[OK]     db
[OK]     slirp
[OK]     logs
[OK]     env
[OK]     vmnetd
[OK]     moby-console
[OK]     moby
[OK]     driver.amd64-linux

Diagnostic ID: 31AFFFAA-CBCC-40E4-8F3D-707D54630FD6
### Steps to reproduce the behavior
1. Run as `docker-compose run --rm --service-ports web /bin/bash`
2. Then `JAMMIT_ENV=staging-ru-edg-reg bundle exec padrino start -h 0.0.0.0 -p 8080` inside
3. After several minutes container crashes with error: `ERROR: dial unix /Users/MY_USERNAME/Library/Containers/com.docker.docker/Data/*00000003.00000948: connect: connection refused`

**Comment by dsheets** [comment]
It looks like the hypervisor is crashing with

```
Docker[com.docker.driver.amd64-linux][437] <Notice>: Assertion failed: (!REPLY_RING_EMPTY(sc)), function send_response_common, file src/pci_virtio_sock.c, line 950.
```

which is similar to, but not the same as, docker/hyperkit#50. There were a number of updates to the hypervisor transport system in the recently released Beta 27. If you are comfortable running a beta (we still do QA on it), you could give that a try and see if it stops the frequent crashing.

/cc @ijc25

**Comment by ijc25** [comment]
This is effectively the same issue as docker/hyperkit#50, the only change is that in 1.12.1 I added some macros with sensible names to make the logic a bit easier to read so the message has changed, but the underlying code's behaviour is identical.

Nonetheless, the request to try beta27 is a good one since that does contain functional changes which should help with that issue.

**Comment by ijc25** [comment]
This was fixed in 1.12.3

**Comment by jcamilli** [comment]
My docker container was crashing on my Mac without an error and after struggling for a while realized that the Docker container JVM was using more resources than allocated by the Mac Docker service. Took me a little too long to figure out this simple problem. No error message at all.

Posting here to potentially help someone else.

<img width="351" alt="screen shot 2018-03-24 at 11 57 08 am" src="https://user-images.githubusercontent.com/1952742/37865990-86baec36-2f5a-11e8-8f8e-4a6866cb4e7f.png">
